irohad/consensus/yac/storage/buffered_cleanup_strategy.hpp

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_BUFFERED_CLEANUP_STRATEGY_HPP
+#define IROHA_BUFFERED_CLEANUP_STRATEGY_HPP
+
+#include "consensus/yac/storage/cleanup_strategy.hpp"
+
+#include <boost/optional.hpp>
+#include <queue>
+
+#include "consensus/yac/outcome_messages.hpp"
+
+namespace iroha {
+  namespace consensus {
+    namespace yac {
+      class BufferedCleanupStrategy : public CleanupStrategy {
+       public:
+        using RoundType = Round;
+
+        /**
+         * The method finalizes passed round. On Commit message it purges last
+         * reject round if commit is greater.
+         * @param consensus_round - finalized round of the consensus
+         * @param answer - the output of the round
+         * @return removed rounds. If rounds are empty returns none
+         */
+        boost::optional<CleanupStrategy::RoundsType> finalize(
+            RoundType consensus_round, Answer answer) override;
+
+        bool shouldCreateRound(const RoundType &round) override;
+
+       private:
+        /**
+         * Remove all rounds before last committed
+         * @return removed rounds
```

Comment:
```suggestion
         * @return rounds to be removed
```

Reply:
https://github.com/hyperledger/iroha/blob/d3e978589a7b0bd2775467c60ef52c3f75e86041/irohad/consensus/yac/storage/impl/buffered_cleanup_strategy.cpp#L50-L51
Technically the rounds are already removed when this method returns a collection

Reply:
But it returns them with a purpose defined at interface level - to tell the yac storage which rounds to remove- it seems more important than the internal strategy state.

Reply:
https://github.com/hyperledger/iroha/blob/d3e978589a7b0bd2775467c60ef52c3f75e86041/irohad/consensus/yac/storage/cleanup_strategy.hpp#L30-L33

Reply:
I preserve both comments